### ISSUE TITLE
Add Single Argument Constructor to ShortcutReader

### ISSUE THREAD

**Issue by madoar** [PhoenicisOrg/phoenicis#2033]
Currently we set the `ShortcutDTO` object of a `ShortcutReader` object via the `of` method. An alternative and in my opinion cleaner approach would be to set the `ShortcutDTO` object via the constructor, especially because a ShortcutReader is unusable without a set `ShortcutDTO` instance. 

The disadvantage of requiring a single argument constructor is that we can't define this in an interface.

For more details see https://github.com/PhoenicisOrg/phoenicis/pull/2025#discussion_r298840320

**Comment by plata** [comment]
It's really a big disadvantage that this cannot be defined in the interface because it practically avoids that anybody can implement the interface correctly without having some hidden knowledge. Therefore I would rather not do this.

**Comment by plata** [comment]
@madoar @qparis can we decide if we want to do this or not?

**Comment by madoar** [comment]
I agree that it is problematic that we can't declare constructor constraints in the interface. I'm fine with retaining it as it is, but I still think that a constructor may look better in the calling code and in the JavaScript class
